.github/workflows/network.yml

```diff
@@ -0,0 +1,70 @@
+name: Network CI
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - 'lib/vdsm/network/**'
+      - 'lib/vdsm/common/network/**'
+      - 'tests/network/**'
+      - '.github/workflows/network.yml'
+  pull_request:
+    paths:
+      - 'lib/vdsm/network/**'
+      - 'lib/vdsm/common/network/**'
+      - 'tests/network/**'
+      - '.github/workflows/network.yml'
+
+jobs:
+  unit:
+    env:
+      IMAGE_TAG: ${{ matrix.tag }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        tag: [ centos-8, centos-9 ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          sudo apt install podman
+      - name: Run unit tests
+        run: sudo -E ./tests/network/unit/run-tests.sh
+
+  integration:
+    env:
+      IMAGE_TAG: ${{ matrix.tag }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        tag: [ centos-8, centos-9 ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          sudo apt install podman
+      - name: Run integration tests
+        run: sudo -E ./tests/network/integration/run-tests.sh
+
+  functional:
+    env:
+      IMAGE_TAG: ${{ matrix.tag }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        tag: [ centos-8, centos-9 ]
+        switch_type: [ ovs, linux-bridge ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          sudo apt install podman
+      - name: Run functional tests
+        run: |
+          sudo -E ./tests/network/functional/run-tests.sh \
+          --switch-type=${{ matrix.switch_type }} \
```

Comment:
Not necessarily for this PR, but given that we pass 'matrix.switch_type' as a cmdline option maybe it would be better to do the same with 'matrix.tag' instead of passing it as env variable.

Reply:
Only issue is that for unit and integration it would require more work as those scripts do not accept cmd options, but definitely possible.